server/controllers/AuthorController.js

```diff
@@ -29,4 +29,50 @@ export default class AuthorController {
       return responseMessage(response, 500, { message: error.message });
     }
   }
+
+  /**
+   * @description allows admin to update an author
+   * @param {object} request express request object
+   * @param {object} response express response object
+   * @returns {json} json
+   * @memberof AuthorControllers
+   */
+  static async updateAuthor(request, response) {
+    const { params: { authorId }, body: { authorName } } = request;
+    try {
+      const existingAuthor = await Author.findOne({ where: { id: authorId } });
+      if (!existingAuthor) { return responseMessage(response, 404, { message: 'author not found' }); }
+      const sameAuthor = await Author.findOne({ where: { fullname: authorName } });
```

Comment:
The logic here is quite wrong. Why query the database to check if that name already exists? The author's fullname isn't a unique attribute, hence, it is possible to actually have two authors with the same name but different IDs. What you should rather do is to check if the fullname of the author with the given authorId is the same as the authorName in the params. This way, the response will tell the user that the fullname for the author with that ID is already the same as the authorName being passed in.

Reply:
done

Reply:
@Ukhu, author's fullname is meant to be unique, why would an author have two ids

Reply:
I didn't say one author should have two IDs. What I am trying to say is that 2 different authors can have the same. There can be two authors named Bella Oyedele. How do you differentiate them? That's why the ID is important. Author fullname can't be unique.